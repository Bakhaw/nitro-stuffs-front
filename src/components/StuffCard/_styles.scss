.StuffCard {
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  &__header {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding: 10px;
    border-radius: 6px 6px 0px 0px;
    background-color: $White;

    span {
      font-size: 16px;
      margin-left: 10px;
    }

    &__title__flashbang {
      color: $Yellow;
    }

    &__title__smoke {
      color: $Purple;
    }

    &__title__molotov {
      color: $Red;
    }
  }

  iframe {
    border-radius: 0 0 6px 6px;
  }
}
