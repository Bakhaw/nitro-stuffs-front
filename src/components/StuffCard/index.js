import React from 'react';
import PropTypes from 'prop-types';

function StuffCard({ stuff }) {
  const { title, type, videoId } = stuff;

  return (
    <div className='StuffCard'>
      <div className='StuffCard__header'>
        <img
          alt={`${type} icon`}
          src={require(`../../static/images/csgo/stuff-icons/${type}.svg`)}
        />
        <span>{title}</span>
      </div>
      <iframe
        allowFullScreen
        frameBorder={0}
        height='170'
        width='100%'
        src={`https://www.youtube.com/embed/${videoId}`}
        title={title}
      />
    </div>
  );
}

StuffCard.propTypes = {
  stuff: PropTypes.object.isRequired
};

export default StuffCard;
