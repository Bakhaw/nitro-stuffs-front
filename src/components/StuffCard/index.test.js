import React from 'react';
import { shallow } from 'enzyme';

import StuffCard from '.';

test('should render properly', () => {
  const stuff =   {
    id: 'id',
    title: 'title',
    type: 'type',
    videoId: 'videoId' // youtube video ID
  },

  const container = shallow(<StuffCard stuff={stuff} />);
  expect(container.debug()).toMatchSnapshot();
});
