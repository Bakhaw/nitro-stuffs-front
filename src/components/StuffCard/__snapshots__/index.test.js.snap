// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"StuffCard\\">
  <div className=\\"StuffCard__header\\">
    <img alt=\\"smoke icon\\" src={{...}} />
    <span>
      title
    </span>
  </div>
  <iframe allowFullScreen={true} frameBorder={0} height=\\"170\\" width=\\"100%\\" src=\\"https://www.youtube.com/embed/videoId\\" title=\\"title\\" />
</div>"
`;
